src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h

```diff
@@ -142,7 +146,8 @@ namespace xclcpuemhal2 {
 
       void set_messagesize( unsigned int messageSize ) { message_size = messageSize; }
       unsigned int get_messagesize(){ return message_size; }
-
+      void setDeviceProcessStarted(bool devProcess) { mIsDeviceProcessStarted = devProcess; }
+      bool getDeviceProcessStarted() const { return mIsDeviceProcessStarted; }
```

Comment:
set and get and class private data member do not go hand in hand.  What's the point of an encapsulated  private data member that can be accessed as if it was public.  The design is wrong, because I am not suggesting you turn `mIsDeviceProcessStarted` into a  public member.

Reply:
Removed set and get. Using mIsDeviceProcessStarted directly.